File: solver.js
"use strict";

const solveQuadratic = (a, b, c) => {
    const d = Math.pow(b, 2) - 4 * a * c;

    if (d > 0) {
        const x1 = (-b + Math.sqrt(d)) / (2 * a);
        const x2 = (-b - Math.sqrt(d)) / (2 * a);
        console.log(`x1 = ${x1}\nx2 = ${x2}`);
    } else if (d === 0) {
        const x = -b / (2 * a);
        console.log(`x = ${x}`);
    } else {
        console.log("There are no roots.")
    }
}

export { solveQuadratic };

File: methodoligies-lab1-swift/methodoligies-lab1-swift/solver.swift
import Foundation

func solveQuadratic(a: Int, b: Int, c: Int) {
    let d = b * b - 4 * a * c
    print("\(a)x^2 + \(b)x + \(c) = 0")
    
    if d > 0 {
        let x1 = (-Double(b) + sqrt(Double(d))) / (2 * Double(a))
        let x2 = (-Double(b) - sqrt(Double(d))) / (2 * Double(a))
        print("x1 = \(x1), x2 = \(x2)")
    } else if d == 0 {
        let x = -Double(b) / (2 * Double(a))
        print("x = \(x)")
    } else {
        print("There are no roots.")
    }
}
